engines/dfc_provider/app/services/dfc_provider/variant_fetcher.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Service used to fetch variants related to an entreprise.
+# It improves maintenance as it is the central point requesting
+# Spree::Varaint inside the DfcProvider engine.
+module DfcProvider
+  class VariantFetcher
```

Comment:
This class is on of the main objective of this PR. The way to fetch data collection is centralize here, so easier to maintain it. Typically it could call an OFN service doing the same thing on OFN side, for the API for example, but I couldn't find it.

Reply:
:-) this is called the repository pattern used in many ORM solutions. I like it! https://martinfowler.com/eaaCatalog/repository.html

AFAIK in rails these are usually scopes in the model itself (something I dont like about rails because scopes overload the models, the repository pattern helps separate scopes from the proper ORM code of the model).

So, this could be a scope in app/model/variant.rb called supplied_by.
If you keep it as a service, I think it should be out of DFC and directly in app/services of the main app.

Reply:
Oh ok, well, I was not pretending to have invented a pattern :) Just add a comment to emphasise the fact that one of the main objective of the PR was here.

I didn't want to touch much the OFN app codebase as I thought discussion was more about extracting this engine inside a separated gem so I didn't want to mess up with a new dependency. We could create new services to help with OFN logic, but maybe not a specific one for DFC, something used by the OFN codebase and then reused inside DFC API. DFC API should just map existing data flow logic somehow, no?